src/routes/intro/ProfessionSelectionScreen.tsx

```diff
@@ -0,0 +1,121 @@
+import { RoutesParams } from '../../navigation/NavigationTypes'
+import AsyncStorage from '../../services/AsyncStorage'
+import { childrenDescription } from '../../services/helpers'
+
+const Root = styled.View`
+  background-color: ${props => props.theme.colors.background};
+  height: 100%;
+`
+
+const ButtonContainer = styled.View`
+  padding-top: ${props => props.theme.spacings.md}
+  margin: 0 auto 0;
+`
+const IconContainer = styled.View`
+  margin-right: ${props => props.theme.spacings.sm};
+`
+const Placeholder = styled.View`
+  width: 24px;
+  margin-right: ${props => props.theme.spacings.sm};
+`
+
+interface ProfessionSelectionScreenProps {
+  route: RouteProp<RoutesParams, 'ProfessionSelection'>
+  navigation: StackNavigationProp<RoutesParams, 'ProfessionSelection'>
+}
+
+const ProfessionSelectionScreen = ({ route, navigation }: ProfessionSelectionScreenProps): JSX.Element => {
+  const { discipline } = route.params
+  const { data: disciplines, error, loading, refresh } = useLoadDisciplines(discipline)
+  const { data: selectedProfessions, refresh: refreshSelectedProfessions } = useReadSelectedProfessions()
+  const isProfessionSelected = selectedProfessions && selectedProfessions.length > 0
+
+  const selectDiscipline = async (selectedItem: Discipline): Promise<void> => {
+    if (selectedProfessions?.some(profession => profession.id === selectedItem.id)) {
+      await AsyncStorage.removeSelectedProfession(selectedItem).then(refreshSelectedProfessions)
+    } else {
+      await AsyncStorage.pushSelectedProfession(selectedItem).then(refreshSelectedProfessions)
+    }
+  }
+
+  const Item = ({ item }: { item: Discipline }): JSX.Element => (
+    <DisciplineListItem
+      item={item}
+      onPress={() => selectDiscipline(item)}
+      hasBadge
+      rightChildren={
+        selectedProfessions?.some(profession => profession.id === item.id) ? (
+          <IconContainer>
+            <CheckCircleIconGreen testID='check-icon' />
+          </IconContainer>
+        ) : (
+          <Placeholder />
+        )
+      }
+    />
+  )
+
+  const navigateToHomeScreen = async () => {
+    if (!isProfessionSelected) {
+      await AsyncStorage.setSelectedProfessions([])
+    }
+    navigation.push('Home')
+  }
+
+  return (
+    <Root>
+      <StatusBar backgroundColor='blue' barStyle='dark-content' />
+      <ServerResponseHandler error={error} loading={loading} refresh={refresh}>
+        <FlatList
+          contentContainerStyle={{ flexGrow: 1 }}
+          ListHeaderComponent={<Title title={discipline.title} description={childrenDescription(discipline)} />}
+          ListFooterComponent={
+            !isProfessionSelected ? (
+              <ButtonContainer>
+                <Button
+                  onPress={navigateToHomeScreen}
+                  label={labels.intro.skipSelection}
+                  buttonTheme={BUTTONS_THEME.contained}
+                />
+              </ButtonContainer>
+            ) : (
+              <></>
```

Comment:
no need to render a fragment if there is no content in it. Instead return `: null`